worlds/sc2/Rules.py

```diff
@@ -416,7 +425,27 @@ def zerg_pass_vents(self, state: CollectionState) -> bool:
     def supreme_requirement(self, state: CollectionState) -> bool:
         return self.story_tech_granted \
             or not self.kerrigan_unit_available \
-            or state.has_all({ItemNames.KERRIGAN_LEAPING_STRIKE, ItemNames.KERRIGAN_MEND}, self.player)
+            or (
+                state.has_all({ItemNames.KERRIGAN_LEAPING_STRIKE, ItemNames.KERRIGAN_MEND}, self.player)
+                and self.kerrigan_levels(state, 35)
```

Comment:
I think this logic requirement can be lowered slightly, but not necessary to do now (30 seems reasonable to me)

Reply:
I'll wait for the feedback